### PR TITLE
update ci-fix 

### PR DESCRIPTION
**Pull request by jianyexi**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: documentation/ci-fix.md
Comment (akning-ms) [inline documentation/ci-fix.md:51]: @JeffreyRichter to comments for below cross API version breaking change detection logic

File: documentation/ci-fix.md
Comment (josefree) [inline documentation/ci-fix.md:52]: Both uses the OAD (OpenAPI Diff) tool to identify breaking changes between swaggers.

File: documentation/ci-fix.md
Comment (josefree) [inline documentation/ci-fix.md:53]: The same version breaking change check compares the existing version swaggers with updated swaggers of the same version, so that to identify if there are any updates break the version.

File: documentation/ci-fix.md
Comment (josefree) [inline documentation/ci-fix.md:54]: The cross version breaking change check usually applies to new API version scenario. It uses swaggers of the latest public version as the baseline, comparing swaggers of new version with the baseline, so that to identify if there are any updates lead to breaking experience cross versions

Comment (josefree) [reply documentation/ci-fix.md:54]: The latest public version can be:

File: documentation/ci-fix.md
Comment (josefree) [inline documentation/ci-fix.md:55]: 1. The latest stable version which released within 3 years.
2. When there is no stable version released within 3 years, then the latest preview version which released within 1 years.

File: documentation/ci-fix.md
Comment (josefree) [inline documentation/ci-fix.md:57]: why 3?

**Review by josefree — CHANGES_REQUESTED** [review]
some wording revises recommended

File: documentation/ci-fix.md
Comment (jianyexi) [reply documentation/ci-fix.md:57]: Because over 1 year preview version needs to GA or retire

File: documentation/ci-fix.md
Comment (raych1) [inline documentation/ci-fix.md:76]: Typo of `current`.

File: documentation/ci-fix.md
Comment (raych1) [inline documentation/ci-fix.md:76]: Typo of last `against`.

File: documentation/ci-fix.md
Comment (JeffreyRichter) [inline documentation/ci-fix.md:53]: The same version check verifies that the API contract has not changed within a single api-version.

File: documentation/ci-fix.md
Comment (JeffreyRichter) [inline documentation/ci-fix.md:54]: The cross version check ensures that a new api-version is backward compatible with the previous api-version.

File: documentation/ci-fix.md
Comment (JeffreyRichter) [inline documentation/ci-fix.md:56]: The latest stable version.
(I would not mention the 3 year thing)

File: documentation/ci-fix.md
Comment (JeffreyRichter) [inline documentation/ci-fix.md:57]: When there is no stable version, the latest preview version that is more than 1 year old.

File: documentation/ci-fix.md
Comment (JeffreyRichter) [inline documentation/ci-fix.md:58]: I think you can delete #3

File: documentation/ci-fix.md
Comment (akning-ms) [reply documentation/ci-fix.md:57]: The real logic is: When there is no stable version(or latest stable version is over 3 year), Then latest preview version can not be break(no matter if over1 year.   maybe Jianye you can re-wording this part)

File: documentation/ci-fix.md
Comment (akning-ms) [reply documentation/ci-fix.md:58]: Here logic is:  if latest preview version is over 1 year. it can not be broken(no matter if there is stable version)
So it should be keep if you are agree above #2 comment . right?

File: documentation/ci-fix.md
Comment (JeffreyRichter) [reply documentation/ci-fix.md:57]: Oh, I think I see what you're going for here. You're saying that it is OK to break if the latest stable is > 3 years old, right? If so, I think this is a bit dangerous and probably shouldn't be a rule. Azure's policy is that customers have to be given 3 years notice before **_forcing_** a breaking version on them. So, a version could be 10 years old but, if we haven't notified customers that the next version is breaking, then the breaking change is NOT ok. In other words, the trigger is notification (after approval); not duration of latest stable version. 

Also, what is more common is for a breaking change to be introduced within 3 years of the last stable and then, with approval & customer notification, this is allowed.

File: documentation/ci-fix.md
Comment (JeffreyRichter) [reply documentation/ci-fix.md:58]: More specifically, a preview cannot break customers using the last GA (without approval).
A preview CAN break the last preview.
We do have a policy that a feature should be in preview for > 1 year but I'm not sure tooling should look for this. For example, what if preview-1 of a feature comes out in January and then preview-2 comes out in December. The 1st preview was 11 months old and then the new preview just started but the feature itself is in preview for >1 year. I don't know how your tooling can detect this. 

Maybe it would be useful to have tooling that periodically scans the repo looking for a preview (with no GA) that has been around for 9 months and then notify the team that they have 3 months left to go GA?

Also, perhaps tooling that finds old preview swaggers that do have GA swagger and tell the team to delete them from the repo so as to not clutter the repo with old preview swaggers and to reduce customer confusion when they see all the swaggers in the repo..

File: documentation/ci-fix.md
Comment (akning-ms) [reply documentation/ci-fix.md:57]: yes, agree with you, Jeffery, As most common case is compare with stable version within 3 year. we can remove this condition(<= 3year) from tool. if so, for the case (> 3 year stable version), if there is breaking change, we will ask service team to request approval from you. :)

File: documentation/ci-fix.md
Comment (akning-ms) [reply documentation/ci-fix.md:58]: We just simply use date in version as time. it maybe not very accurate.  yes, it seems a good idea to scan repo, we will discuss it further internally then get back to you.

File: documentation/ci-fix.md
Comment (jianyexi) [reply documentation/ci-fix.md:57]: @JeffreyRichter any concern about the above proposal by @akning-ms

**Review by josefree — APPROVED** [review]

File: documentation/ci-fix.md
Comment (jianyexi) [reply documentation/ci-fix.md]: already synced offline

**Pull request merged**
